zeppelin-zengine/src/main/java/org/apache/zeppelin/helium/HeliumOnlineRegistry.java

```diff
@@ -55,7 +56,10 @@
   public HeliumOnlineRegistry(String name, String uri, File registryCacheDir) {
     super(name, uri);
     registryCacheDir.mkdirs();
-    this.registryCacheFile = new File(registryCacheDir, name);
+
+    UUID registryCacheFileUuid = UUID.nameUUIDFromBytes(uri.getBytes());
```

Comment:
Yes, that was the intention to use `namedUUIDFromBytes` instead of `randomUUID`. But `randomUUID` would also work if the cache dir is purged (but I was not sure if that is guaranteed).

To be precise: nameUUIDFromBytes implement variant 3 from http://www.ietf.org/rfc/rfc4122.txt:

> 4.3.  Algorithm for Creating a Name-Based UUID
>
>    The version 3 or 5 UUID is meant for generating UUIDs from "names"
>    that are drawn from, and unique within, some "name space".  The
>    concept of name and name space should be broadly construed, and not
>    limited to textual names.  For example, some name spaces are the
>    domain name system, URLs, ISO Object IDs (OIDs), X.500 Distinguished
>    Names (DNs), and reserved words in a programming language.  The
>    mechanisms or conventions used for allocating names and ensuring
>    their uniqueness within their name spaces are beyond the scope of
>    this specification.
> 
>    The requirements for these types of UUIDs are as follows:
> 
>    o  The UUIDs generated at different times from the same name in the
>       same namespace MUST be equal.
> 
>    o  The UUIDs generated from two different names in the same namespace
>       should be different (with very high probability).
> 
>    o  The UUIDs generated from the same name in two different namespaces
>       should be different with (very high probability).
> 
>    o  If two UUIDs that were generated from names are equal, then they
>       were generated from the same name in the same namespace (with very
>       high probability).

If we would like to ensure different names, we should use randomUUID, but then the cache dir will pile up new cache files unless it is purged on a regular basis. Since I was not sure if that is the case, I proposed the "hash-like" nameUUIDFromBytes.